src/mercury_engine_data_structures/adapters/flagsenum_adapter.py

```diff
@@ -0,0 +1,15 @@
+from __future__ import annotations
+
+from construct import Adapter, FlagsEnum, Int32ub
+
+
+class FlagsEnumAdapter(Adapter):
+    def __init__(self, enum_class, subcon=Int32ub):
+        super().__init__(FlagsEnum(subcon, enum_class))
+        self._enum_class = enum_class
+
+    def _decode(self, obj, context, path):
+        return {self._enum_class[k]: v for k, v in obj.items() if k != "_flagsenum" and v is True}
+
+    def _encode(self, obj, context, path):
+        return obj
```

Comment:
ah, looks like this doesn't work. try this instead: `return {k.name: v for k, v in obj.items()}`